pkg/compact/compact.go

```diff
@@ -907,6 +930,12 @@ func (c *BucketCompactor) Compact(ctx context.Context) error {
 			return errors.Wrap(err, "compaction")
 		}
 		if done {
+			// Process index cache if only all compaction done at that moment.
+			level.Info(c.logger).Log("msg", "process index cache files")
+			if err := c.sy.IndexCache(ctx); err != nil {
```

Comment:
Again, as per https://github.com/improbable-eng/thanos/pull/986#discussion_r270219562 why we are not doing this:
* in sync? It's not related to sync at all.
* After all groups are done? It would better tied to compactor process... (: As we can do in concurrently at some point (different groups), like this:  https://github.com/improbable-eng/thanos/pull/1010

Reply:
Are u suggesting exposing blocks from Sync and pass them to CacheProcessor?

>As we can do in concurrently at some point (different groups), like this: #1010

this is one-time task, don't see any reason to make this complicated, it will change cpu/memory footprint if you don it concurrently.

Reply:
So this comment relates to two different things, let's separate those:

> After all groups are done? It would better tied to compactor process... (: As we can do in concurrently at some point (different groups), like this: #1010

Reply:
>Well, you don't need to do anything. After sync you already have blocks (at least meta files), so I would vote to have a function for group and run it if generateMissingIndexCacheFiles somewhere in the group loop. This will nicely go with concurrency as planned here: #1010 (actually merged) Also this relates to ONLY this temporary function that loops over group blocks and see if there is missing index cache file!

makes sense, will change

>Yes, I think we are missing long term solution here, so after compaction we would love to create the index cache file the new block before removing old blocks (: Can we do it here?

no needs to bring it here, it's already a part of compaction/downsampling proccess
https://github.com/improbable-eng/thanos/pull/986/files#diff-921a1164068d3a04217b2d85714b16adR829

https://github.com/improbable-eng/thanos/pull/986/files#diff-059ebb3a930d53936a164ef53a79d7f0R198